### PR TITLE
[WIN32K:NTUSER] Update USER_TIMER_M*IMUM handling to current target

### PR DESCRIPTION
**Pull request by SergeGautherie**
which is WS03 SP2.

### CODE REVIEW

File: win32ss/user/ntuser/sysparams.c [hunk 1]
```diff
@@ -1490,14 +1490,19 @@ SpiGetSet(UINT uiAction, UINT uiParam, PVOID pvParam, FLONG fl)
...
+        {
+            /* Windows XP SP2 and Windows Server 2003 SP1 new behaviour */
+            if (uiParam < USER_TIMER_MINIMUM)
+            {
+               uiParam = USER_TIMER_MINIMUM;
+            }
+            else if (uiParam > USER_TIMER_MAXIMUM)
+            {
+               uiParam = USER_TIMER_MAXIMUM;
+            }
```
Comment (HBelusca) [inline win32ss/user/ntuser/sysparams.c:1502]:
```suggestion
            uiParam = min(max(uiParam, USER_TIMER_MINIMUM), USER_TIMER_MAXIMUM);
```

File: win32ss/user/ntuser/timer.c [hunk 2]
```diff
@@ -185,28 +185,18 @@ IntSetTimer( PWND Window,
...
-     TRACE("Adjusting uElapse\n");
+     TRACE("Adjusting uElapse to USER_TIMER_MINIMUM\n");
      Elapse = USER_TIMER_MINIMUM; // 1024hz .9765625 ms, set to 10.0 ms (+/-)1 ms
   }
+  else if (Elapse > USER_TIMER_MAXIMUM)
+  {
+     /* Windows XP SP2 and Windows Server 2003 new behaviour */
+     TRACE("Adjusting uElapse to USER_TIMER_MAXIMUM\n");
+     Elapse = USER_TIMER_MAXIMUM;
+  }
```
Comment (HBelusca) [inline win32ss/user/ntuser/timer.c:199]: Not sure we really care about these traces, in which case doing a
```Elapse = min(max(Elapse, USER_TIMER_MINIMUM), USER_TIMER_MAXIMUM);```
would be more "straightforward".

**Comment by HBelusca** [comment]
Otherwise you can use things like:
```#if (NTDDI_VERSION >= NTDDI_WS03) ... #endif```; that would convey the very same information as the added comments, and in a compile-time manner.

File: win32ss/user/ntuser/sysparams.c (hunk 1 above)
Comment (SergeGautherie) [reply win32ss/user/ntuser/sysparams.c:1502]: I am not immediately convinced: good as 1-liner, but `min` and `max` never apply together.

File: win32ss/user/ntuser/timer.c (hunk 2 above)
Comment (SergeGautherie) [reply win32ss/user/ntuser/timer.c:199]: TRACE: I don't know, so I am not removing them (yet).
min/max: See reply to other comment.

**Review by SergeGautherie — COMMENTED** [review]
> `NTDDI_WS03

That does not seem wanted, as old behaviour shall never be used again.

**Comment by HBelusca** [comment]
What do you mean by "min and max do not apply together" ?

**Comment by SergeGautherie** [comment]
> What do you mean by "min and max do not apply together" ?

A value can't be both too low and too high.

**Comment by HBelusca** [comment]
> > What do you mean by "min and max do not apply together" ?
> 
> A value can't be both too low and too high.

Then you haven't understood what the min and max macros do.

**Comment by SergeGautherie** [comment]
> you haven't understood

Really? Please, explain.

**Comment by ThFabba** [comment]
> > you haven't understood
> 
> Really? Please, explain.

This is tiresome.
Hermès provided a code sample, you're claiming it doesn't work. Prove it.

**Comment by SergeGautherie** [comment]
> you're claiming it doesn't work

No, I do not: I simply wonder why "max and min" is better than "if_< or if_>" in this case(s).

**Pull request closed without merging**